app/Models/Team.php

```diff
@@ -76,6 +76,19 @@ public function descriptionHtml(): string
             : bbcode((new BBCodeForDB($description))->generate());
     }
 
+    public function delete()
+    {
+        $ret = parent::delete();
+
+        if ($ret) {
+            $this->header()->delete();
+            $this->logo()->delete();
+            $this->members()->delete();
```

Comment:
I feel like the db stuff should be deleted together in a transaction, and the assets should be removed first so there's no dangling stuff if something somehow fails (can just try delete the team again)